cmd/create/oidcconfig/cmd.go

```diff
@@ -136,55 +174,62 @@ const (
 	jwksKey              = "keys.json"
 )
 
-func (s *CreateOidcConfigAutoStrategy) execute(r *rosa.Runtime) (string, string) {
-	randomLabel := helper.RandomLabel(DEFAULT_LENGTH_RANDOM_LABEL)
-	bucketName := fmt.Sprintf("oidc-%s", randomLabel)
-	bucketUrl := fmt.Sprintf("https://%s.s3.%s.amazonaws.com", bucketName, args.region)
-	err := r.AWSClient.CreateS3Bucket(bucketName, args.region)
+func (s *CreateOidcConfigAutoStrategy) execute(r *rosa.Runtime, input OidcConfigInput) {
+	bucketName := input.BucketName
+	discoveryDocument := input.DiscoveryDocument
+	jwks := input.Jwks
+	privateKey := input.PrivateKey
+	err := r.AWSClient.CreateSecretInSecretsManager(bucketName, string(privateKey[:]))
 	if err != nil {
-		r.Reporter.Errorf("There was a problem creating S3 bucket '%s': %s", bucketName, err)
+		r.Reporter.Errorf("There was a problem saving private key to secrets manager: %s", err)
 		os.Exit(1)
 	}
-	privateKey, publicKey, err := createKeyPair()
+	err = r.AWSClient.CreateS3Bucket(bucketName, args.region)
 	if err != nil {
-		r.Reporter.Errorf("There was a problem generating key pair: %s", err)
-		os.Exit(1)
-	}
-	privateKeyFilename := fmt.Sprintf("private-key-%s.key", bucketName)
-	err = helper.SaveDocument(string(privateKey[:]), privateKeyFilename)
-	if err != nil {
-		r.Reporter.Errorf("There was a problem saving private key to a file: %s", err)
+		r.Reporter.Errorf("There was a problem creating S3 bucket '%s': %s", bucketName, err)
 		os.Exit(1)
 	}
-	discoveryDocument := generateDiscoveryDocument(bucketUrl)
 	err = r.AWSClient.PutPublicReadObjectInS3Bucket(
 		bucketName, strings.NewReader(discoveryDocument), discoveryDocumentKey)
 	if err != nil {
 		r.Reporter.Errorf("There was a problem populating discovery "+
 			"document to S3 bucket '%s': %s", bucketName, err)
 		os.Exit(1)
 	}
-	jwks, err := buildJSONWebKeySet(publicKey)
-	if err != nil {
-		r.Reporter.Errorf("There was a problem generating JSON Web Key Set: %s", err)
-		os.Exit(1)
-	}
 	err = r.AWSClient.PutPublicReadObjectInS3Bucket(bucketName, bytes.NewReader(jwks), jwksKey)
 	if err != nil {
 		r.Reporter.Errorf("There was a problem populating JWKS "+
 			"to S3 bucket '%s': %s", bucketName, err)
 		os.Exit(1)
 	}
-	return bucketUrl, privateKeyFilename
 }
 
 type CreateOidcConfigManualStrategy struct{}
 
-func (s *CreateOidcConfigManualStrategy) execute(r *rosa.Runtime) (string, string) {
+func (s *CreateOidcConfigManualStrategy) execute(r *rosa.Runtime, input OidcConfigInput) {
 	commands := []string{}
-	randomLabel := helper.RandomLabel(DEFAULT_LENGTH_RANDOM_LABEL)
-	bucketName := fmt.Sprintf("oidc-%s", randomLabel)
-	bucketUrl := fmt.Sprintf("https://%s.s3.%s.amazonaws.com", bucketName, args.region)
+	bucketName := input.BucketName
+	discoveryDocument := input.DiscoveryDocument
+	jwks := input.Jwks
+	privateKey := input.PrivateKey
+	privateKeyFilename := input.PrivateKeyFilename
```

Comment:
I see... You're storing the file for manual mode? Could be useful, but I wouldn't invest too much in this for the time being. That said, if it all works well then keep the code in place.

Reply:
We need to keep it for manual as create secret requires a file, I mean not requires but it is safer than output the private key in the screen to run the command. It would also generate a big command which is difficult to see where it ends

Within the listing of commands I also add commands to remove the files that have been generated